store/mongo/datastore_mongo_test.go

```diff
@@ -1378,6 +1378,54 @@ func TestStoreAuthSet(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestUpsertAuthSet(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping TestGetDevices in short mode.")
+	}
+
+	ctx := context.Background()
+	ctx = identity.WithContext(ctx, &identity.Identity{Tenant: "1234"})
+	db := getDb(ctx)
+
+	authSet := &model.AuthSet{
+		IdData:       `{"foo":"bar"}`,
+		IdDataSha256: []byte("7a38bf81f383f69433ad6e900d35b3e2385593f76a7b7ab5d4355b8ba41ee24b"),
+		PubKey: `-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAE40NzSJ0FefAHPTDR5eUdEiTIhcztPX58
+/YPgwRoRydYGw299yP5uH+SEHOZUdIxcTLAvGQME5evgPxNjVmq/NVIeXZunTcYs
+UTnpeomhxO70PoGy64hqCAp4cIGDrVJT
+-----END PUBLIC KEY-----`,
+		Status: "pending",
+	}
+
+	err := db.UpsertAuthSetStatus(ctx, authSet)
+	assert.NoError(t, err)
+
+	newPreauth := &model.AuthSet{
+		Id:           "boink!",
+		IdData:       `{"foo":"bar"}`,
+		IdDataSha256: []byte("7a38bf81f383f69433ad6e900d35b3e2385593f76a7b7ab5d4355b8ba41ee24b"),
+		PubKey: `-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAE40NzSJ0FefAHPTDR5eUdEiTIhcztPX58
+/YPgwRoRydYGw299yP5uH+SEHOZUdIxcTLAvGQME5evgPxNjVmq/NVIeXZunTcYs
+UTnpeomhxO70PoGy64hqCAp4cIGDrVJT
+-----END PUBLIC KEY-----`,
+		Status: "preauthorized",
+	}
+
```

Comment:
please add an assert that and `newPreauth` and `authSet` are equal up to the status.

Reply:
I believe this is the assertion made here: https://github.com/mendersoftware/deviceauth/blob/8f12c84773a3aba9d9b4b4878b1b967cf573d327/store/mongo/datastore_mongo_test.go#L1425
I'll tidy up the test a little and add some comments to make it easier to read.